lib/rubocop/cop/rspec/capybara/negation_matcher.rb

```diff
@@ -0,0 +1,104 @@
+    module RSpec
+      module Capybara
+        # Enforces use of `have_no_*` or `not_to` for negated expectations.
+        #
+        # @example EnforcedStyle: have_no (default)
+        #   # bad
+        #   expect(page).not_to have_selector
+        #   expect(page).not_to have_css('a')
+        #
+        #   # good
+        #   expect(page).to have_no_selector
+        #   expect(page).to have_no_css('a')
+        #
+        # @example EnforcedStyle: not_to
+        #   # bad
+        #   expect(page).to have_no_selector
+        #   expect(page).to have_no_css('a')
+        #
+        #   # good
+        #   expect(page).not_to have_selector
+        #   expect(page).not_to have_css('a')
+        #
+        class NegationMatcher < Base
+          extend AutoCorrector
+          include ConfigurableEnforcedStyle
+
+          MSG = 'Use `%<good>s` instead of `%<bad>s`.'
+          CAPYBARA_MATCHERS = %w[
+            selector css xpath text title current_path link button
+            field checked_field unchecked_field select table
+            sibling ancestor
+          ].freeze
+          POSITIVE_MATCHERS = CAPYBARA_MATCHERS.map do |element|
+            "have_#{element}".to_sym
+          end.freeze
+          NEGATIVE_MATCHERS = CAPYBARA_MATCHERS.map do |element|
+            "have_no_#{element}".to_sym
+          end.freeze
+          RESTRICT_ON_SEND = (POSITIVE_MATCHERS + NEGATIVE_MATCHERS).freeze
+
+          # @!method not_to?(node)
+          def_node_matcher :not_to?, <<~PATTERN
+            (send ... :not_to
+              (send nil? {#{POSITIVE_MATCHERS.map(&:inspect).join(' ')}} ...))
+          PATTERN
+
+          # @!method have_no?(node)
+          def_node_matcher :have_no?, <<~PATTERN
+            (send ... :to
+              (send nil? {#{NEGATIVE_MATCHERS.map(&:inspect).join(' ')}} ...))
+          PATTERN
+
+          def on_send(node)
+            return unless offense?(node.parent)
+
+            add_offense(offense_range(node),
+                        message: message(node)) do |corrector|
+              corrector.replace(node.parent.loc.selector, replaced_target)
+              corrector.replace(node.loc.selector, replaced_matcher(node))
+            end
+          end
+
+          private
+
+          def offense?(node)
+            (style == :have_no && not_to?(node)) ||
+              (style == :not_to && have_no?(node))
+          end
+
+          def offense_range(node)
+            node.parent.loc.selector.with(end_pos: node.loc.selector.end_pos)
+          end
+
+          def message(node)
+            format(MSG,
+                   good: format_message(node.parent.loc.selector.source,
+                                        node.loc.selector.source),
+                   bad: format_message(replaced_target, replaced_matcher(node)))
+          end
+
+          def format_message(target, matcher)
+            "expect(...).#{target} #{matcher}"
+          end
+
+          def replaced_target
```

Comment:
We refer to this a "runner" in `RuboCop::RSpec::Language`. I'm not aware if there's an official RSpec term for `to`/`not_to`.

Reply:
I see. I named this place `target` because `to`/`not_to` belongs to `ExpectationTarget`.
If we are going to name it `runner` I would like to change it, but I would like to hear your opinion.

https://www.rubydoc.info/gems/rspec-expectations/2.14.4/RSpec/Expectations/ExpectationTarget

Reply:
```ruby
expect(something) # => ExpectationTarget wrapping something
```

I agree that the receiver of `to`/`not_to` is an expectation target. However, what is the `to`/`not_to` itself? From [its description](https://www.rubydoc.info/gems/rspec-expectations/2.14.4/RSpec%2FExpectations%2FExpectationTarget:to):
> #to(matcher = nil, message = nil, &block) ⇒ Boolean
> Runs the given expectation, passing if matcher returns true.

i'd say it's a runner if it runs something. However, it's not referenced as such anywhere.

Furthermore, it might add confusion with an unrelated `RSpec::Core::Runner`.

My only point is the internal consistency, since we call it runner in `RuboCop::RSpec::Language`, possibly by mistake.

Leaving this up to you to decide, I'm fine either way.

Reply:
Indeed it is. I will change it to `runner` as I think it is better to keep internal consistency.